tools/cpp-repro-gen.py

```diff
@@ -81,6 +81,7 @@ def handle_call(l: ast.Call):
             func.id = func.id + "<Double>"
         elif isinstance(arg, int):
             func.id = func.id + "<Int>"
+        keywords = []
```

Comment:
Could we have a `Double` scalar with dtype float? I think we should keep the dtype, but as a positional argument.

Reply:
I think we have been using `Double` scalar with dtype `Float` in our code in many cases. I remember when I was doing the reduction PR you mensioned that this is because
```
Python scalar only has double, int64_t and complex double, there is no float, int32 and complex float.
PyTorch scalar has the same design as python scalar, so the dtype might be mismatch here
```

Reply:
I believe at some point @naoyam added support for `Double` scalar with dtype `Float`. Not sure if this is exposed in Python.

Reply:
So if the scalar is a runtime value user provided from Python, then sure, it can not be Float. But if it is a constant, then there is no reason for having to be Double?

Reply:
@liqiangxl Are you updating this?

Reply:
Updated. With the new change given:
```
S20 = fd.define_scalar(20480.0, dtype=DataType.Double)
S20 = fd.define_scalar(20480.0, dtype=DataType.Float)
```
will generate:
```
    auto s20 = IrBuilder::create<Double>(20480.0, DataType::Double);
    auto s20 = IrBuilder::create<Double>(20480.0, DataType::Float);
```